### DISCUSSION TITLE
Should crane recommend using `builtins.path` in the examples?

### DISCUSSION THREAD

**Discussion by dpc** [ipetkov/crane#269] [General]
One of our contributors pointed out that they didn't get cached artifacts from cachix as they were using a different project name than what's the default, and pointed out to:

https://nix.dev/anti-patterns/language#reproducibility-referencing-top-level-directory-with

I'm going change our extra-filtering functions, but I wonder if crane should use it e.g. in the examples (https://crane.dev/examples/quick-start.html). I also wonder - if my code uses `src = ./.;` could crane "wrap it" with `builtins.path` internally? Does it already? Should it?

**Comment by ipetkov** [comment]
TIL `./.` is not always reproducible... But yes this is a great idea!

While we're at it, maybe we should add a `craneLib.path` helper which takes a path and calls `builtins.path`, setting `name` to the workspace name (or `"source"` if no name is set). This would make it slightly less verbose than filling out `builtins.path` directly

**Comment by ipetkov** [comment]
Opened https://github.com/ipetkov/crane/issues/272 to track this, so we can move the discussion there!
